### DISCUSSION TITLE
overriding config

### DISCUSSION THREAD

**Discussion by birojnayak** [dotnet/dotnet-monitor#2135] [Q&A] [answered]
Installed the tool, crated a file at $HOME/.config/dotnet-monitor/settings.json and $XDG_CONFIG_HOME/dotnet-monitor/settings.json (just to ensure it picks one of them). Ran 
1) dotnet-monitor config show , it's still showing the default configuration
2)ran dotnet-monitor collect, still using the old end point port.

How to change the settings.json ? By the way, when I installed in my mac, settings.json was not created, I had to create.

Any guidance on above would help.

**Comment by jander-msft** [comment]
> Installed the tool, crated a file at $HOME/.config/dotnet-monitor/settings.json and $XDG_CONFIG_HOME/dotnet-monitor/settings.json (just to ensure it picks one of them). 

On first inspection, either path looks correct. Could you share the contents of the settings.json file? Feel free to redact or remove parts that may contain sensitive information, such as account keys or user names.

> How to change the settings.json ? By the way, when I installed in my mac, settings.json was not created, I had to create.

The dotnet-monitor tool will not create the configuration files automatically. You have to create them if you want to provide configuration via whichever configuration mechanism you want to use.

**Reply by birojnayak** [reply]
{
  "Authentication": {
    "MonitorApiKey": {
      "PublicKey": "eyfoobar",
      "Subject": "b3913c32-barfoo"
    }
  }
}

**Reply by birojnayak** [reply]
the window path works... fyi..

**Comment by birojnayak** [comment, marked as answer]
closing it... got it working..
